app/models/source_set.rb

```diff
@@ -34,14 +33,27 @@ def self.unpublished_sets
   end
 
   ##
+  # Order SourceSets by a given parameter.
+  # If a parameter is not included in the sort_params hash, it will be ignored.
+  # By default, it will order by 'recently_added'.
+  # @param order String or nil
+  # @return ActiveRecord::Relation
+  def self.order_by(order)
+    sort_params = { 'recently_added' => { created_at: :desc },
+                    'chronology_desc' => { year: :desc },
+                    'chronology_asc' => { year: :asc } }
+    order(sort_params[order] || sort_params['recently_added'])
+  end
+
+  ##
   # Get SourceSets associated with all of the specified tags.
   # EACH returned SourceSet will have ALL of the tags.
   # If no tags are specified, return all SourceSets.
-  # @param tags [Array<String>]
+  # @param tags [Array<String>] or nil
   # @return [Array<SourceSet>]
   def self.with_tags(tags)
     return all unless tags.present?
-    tags.map { |tag| with_tag(tag) }.inject(:&)
+    tags.map { |tag| with_tag(tag) }.inject(:&).to_a
```

Comment:
This is just correcting a bug I found.  Without `to_a` the method may return either an `Array` or an `ActiveRecord::Relation`.  This way, it always returns an `Array`.